lib/agentx.c

```diff
@@ -259,6 +262,16 @@ void smux_init(struct thread_master *tm)
 	install_element(CONFIG_NODE, &no_agentx_cmd);
 }
 
+void smux_agentx_enable(void)
+{
+	if (!agentx_enabled) {
+		init_snmp(FRR_SMUX_NAME);
+		events = list_new();
+		agentx_events_update();
+		agentx_enabled = 1;
+	}
+}
+
```

Comment:
can you confirm.
the ldp process that needs to register will register to that smux_agentx_enable() hook and will automatically be called back?

Reply:
Yes, that's right.

In the LDP SNMP PR, LDP will register for that hook.  When agentx is configured at the CLI, LDP will process the hook callback.